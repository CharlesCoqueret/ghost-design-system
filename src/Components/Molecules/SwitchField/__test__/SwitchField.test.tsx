import React from 'react';
import { render } from '@testing-library/react';

import { SwitchField } from '..';

describe('SwitchField Component', () => {
  it('SwitchField renders', () => {
    const { container } = render(<SwitchField inputValue={[{ label: 'Label', value: 'VALUE' }]} />);
    expect(container).toMatchSnapshot();
  });

  it('SwitchField renders with values in readonly', () => {
    const { container } = render(<SwitchField inputValue={[{ label: 'Label', value: 'VALUE' }]} readOnly />);
    expect(container).toMatchSnapshot();
  });

  it('SwitchField renders with values in disabled highligted', () => {
    const { container } = render(
      <SwitchField inputValue={[{ label: 'Label', value: 'VALUE' }]} disabled highlighted />,
    );
    expect(container).toMatchSnapshot();
  });

  it('SwitchField renders with values with fieldSize and inline', () => {
    const { container } = render(
      <SwitchField inputValue={[{ label: 'Label', value: 'VALUE' }]} inline fieldSize={6} />,
    );
    expect(container).toMatchSnapshot();
  });
});
