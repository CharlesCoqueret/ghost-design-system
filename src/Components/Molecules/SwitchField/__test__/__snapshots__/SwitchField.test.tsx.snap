// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchField Component SwitchField renders 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="container"
    >
      <label
        class="label"
        for="switch-1"
        tabindex="0"
      >
        <div
          class="switch"
        >
          <input
            aria-hidden="true"
            readonly=""
            type="checkbox"
          />
          <span
            aria-label="Label"
            class="pebble"
            id="switch-1"
            role="checkbox"
          />
        </div>
        Label
      </label>
    </div>
  </div>
</div>
`;

exports[`SwitchField Component SwitchField renders with values in disabled highligted 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="container"
    >
      <label
        class="label disabled"
        for="switch-3"
        tabindex="-1"
      >
        <div
          class="switch"
        >
          <input
            aria-hidden="true"
            disabled=""
            readonly=""
            type="checkbox"
          />
          <span
            aria-label="Label"
            class="silver"
            id="switch-3"
            role="checkbox"
          />
        </div>
        Label
      </label>
    </div>
  </div>
</div>
`;

exports[`SwitchField Component SwitchField renders with values in readonly 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="container"
    >
      <label
        class="label readOnly"
        for="switch-2"
        tabindex="-1"
      >
        <div
          class="switch"
        >
          <input
            aria-hidden="true"
            readonly=""
            type="checkbox"
          />
          <span
            aria-label="Label"
            class="silver"
            id="switch-2"
            role="checkbox"
          />
        </div>
        Label
      </label>
    </div>
  </div>
</div>
`;

exports[`SwitchField Component SwitchField renders with values with fieldSize and inline 1`] = `
<div>
  <div
    class="fieldGroup size-6 inline"
  >
    <div
      class="container inlineContainer"
    >
      <label
        class="label inlineLabel"
        for="switch-4"
        tabindex="0"
      >
        <div
          class="switch"
        >
          <input
            aria-hidden="true"
            readonly=""
            type="checkbox"
          />
          <span
            aria-label="Label"
            class="pebble"
            id="switch-4"
            role="checkbox"
          />
        </div>
        Label
      </label>
    </div>
  </div>
</div>
`;
